### PR TITLE
Uplift editor for specifying external format parameter data in service tests

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Fixes https://github.com/finos/legend-studio/issues/2255

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

Before: 

Screenshot 2023-05-31 at 5 36 14 PM

After:

https://github.com/finos/legend-studio/assets/73408381/79dfc8a9-5d17-4a05-9fa3-245f3c127c82

Screenshot 2023-06-05 at 10 35 04 AM

Screenshot 2023-06-05 at 10 35 42 AM

Multiple params:

Screenshot 2023-06-02 at 10 38 21 AM

Screenshot 2023-06-05 at 10 37 15 AM

### CODE REVIEW

**Comment by hardikmaheshwari** [comment]
@YannanGao-gs thanks for adding this support. Wanted to check if we are testing also for Byte[*] type queries and wrap the data into toBytes(<data>) 

Also a minor suggestion, would it be possible to have a button which brings the editor as pop-up on complete screen as normally test would be bigger so that might give a better view. To be clear what I see is already a great uplift from current state so totally fine if its handled later

**Comment by YannanGao-gs** [comment]
> @YannanGao-gs thanks for adding this support. Wanted to check if we are testing also for Byte[*] type queries and wrap the data into toBytes()
> 
> Also a minor suggestion, would it be possible to have a button which brings the editor as pop-up on complete screen as normally test would be bigger so that might give a better view. To be clear what I see is already a great uplift from current state so totally fine if its handled later

Hello @hardikmaheshwari, no problem at all! I only handled `string` for now. Would you mind giving me an example of `byte`.
For multi parameters, i handled this case.

File: packages/legend-query-builder/src/graph/QueryBuilderMetaModelConst.ts [hunk 1]
```diff
@@ -142,4 +142,11 @@ export enum QUERY_BUILDER_SUPPORTED_FUNCTIONS {
   OLAP_AVERAGE_RANK = 'meta::pure::functions::math::olap::averageRank',
   OLAP_DENSE_RANK = 'meta::pure::functions::math::olap::denseRank',
   OLAP_ROW_NUMBER = 'meta::pure::functions::math::olap::rowNumber',
+
+  // External Format
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/graph/QueryBuilderMetaModelConst.ts:146]: we should use the full path for these and leverage the function we have in query-builder to check if the function name is the function

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts [hunk 2]
```diff
@@ -221,6 +235,94 @@ export class ServiceTestSetupState {
...
+
+        if (valueSpec instanceof LambdaFunctionInstanceValue) {
+          return this.getBindingWithParamRecursively(
+            valueSpec.values[0]?.expressionSequence[0],
+          );
+        }
+      } catch (error) {
+        this.editorStore.applicationStore.logService.error(
+          LogEvent.create(
+            LEGEND_STUDIO_APP_EVENT.TEST_DATA_GENERATION__FAILURE,
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts:263]: should this really be logged as test data generation failure ?

File: packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx [hunk 3]
```diff
@@ -338,6 +459,14 @@ const ServiceTestSetupEditor = observer(
                         isReadOnly={isReadOnly}
                         paramState={paramState}
                         serviceTestState={serviceTestState}
+                        bindingParamPair={
+                          setupState
+                            .getBindingWithParamFromQuery()
+                            .filter(
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx:465]: use find instead here

File: packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx [hunk 4]
```diff
@@ -143,48 +180,132 @@ const ServiceTestParameterEditor = observer(
...
+                    <CodeEditor
+                      key={paramState.uuid}
+                      inputValue={
+                        (paramState.valueSpec as PrimitiveInstanceValue)
+                          .values[0] as string
+                      }
+                      updateInput={updateParamValue}
+                      isReadOnly={isReadOnly}
+                      language={
+                        bindingParamPair.binding.contentType ===
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx:203]: dont we have a function for this already ? in binding editor is this what we use ?

**Review by MauricioUyaguari — COMMENTED** [review]
If its not too much could we make the flow a bit more simple. 
Keep the orginal input and add a pop icon on the right. (i think i mispoke and said expand which is what you did :) ). the expand icon would bring up the modal with the formatting needed <CodeEditor> and only enabled if we were able to find the assoicated binding. 
Don't think this is much but happy to move along the PR if it is. just address minor comments on here.

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts (hunk 2 above)
Comment (YannanGao-gs) [reply packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts:264]: Created a new entry  `TEST_DATA_GENERATION__SETUP__FAILURE = 'editor.test.test-data-generation-setup.failure'` 
Let me know if this is better

**Comment by YannanGao-gs** [comment]
> If its not too much could we make the flow a bit more simple. Keep the orginal input and add a pop icon on the right. (i think i mispoke and said expand which is what you did :) ). the expand icon would bring up the modal with the formatting needed and only enabled if we were able to find the assoicated binding. Don't think this is much but happy to move along the PR if it is. just address minor comments on here.

I updated the design and attached new screenshots and videos. Please review. Thank you!

**Review by MauricioUyaguari — APPROVED** [review]

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts (hunk 2 above)
Comment (MauricioUyaguari) [reply packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts]: my point was that the function as nothing to do with test data generation. It is how you define the service test parameters values. So like test setup maybe but nothing to do with generation.

File: packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx (hunk 4 above)
Comment (MauricioUyaguari) [reply packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx]: we should maybe move this logic in a central location getCodeEditorByContentType
can you please do this.

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts [hunk 5]
```diff
@@ -221,6 +236,105 @@ export class ServiceTestSetupState {
     }));
   }
 
+  getBindingWithParamFromQuery(): {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestEditorState.ts:239]: lets add unit tests for this logic.
Thanks!

**Pull request merged**